cogs/commands/roomsearch.py

```diff
@@ -43,76 +76,158 @@ async def room(self, ctx: Context, name: str):
+            )
 
-        await ctx.reply(embed=embed, file=img)
+            await ctx.reply(embed=embed, file=img)
 
     async def choose_room(self, ctx, rooms):
+        # Confirms room choice, esp. important with autocomplete api
+        # Create and send message
         emojis = self.full_emojis[: len(rooms)]
         header = "Multiple rooms exist with that name. Which do you want?:"
         rooms_text = "".join(
-            f"\n\t{e} {r.get('name')} in **{r.get('building')}** {r.get('floor')}"
+            f"\n\t{e} {r.get('value')} in **{r.get('building')}** {r.get('floor')}"
             for r, e in zip(rooms, emojis)
         )
         conf_message = await ctx.send(header + rooms_text)
-        for e in emojis:
-            await conf_message.add_reaction(e)
-
-        try:
-            check = (
-                lambda r, u: r.message.id == conf_message.id
-                and u == ctx.message.author
-                and str(r.emoji) in emojis
-            )
-            react_emoji, _ = await ctx.bot.wait_for(
-                "reaction_add", check=check, timeout=60
-            )
-        except TimeoutError:
-            return None
 
+        # Add reacts to choice msg
+        async def add_reacts():
+            try:
+                for e in emojis:
+                    await conf_message.add_reaction(e)
+            except discord.NotFound:
+                pass
+
+        # Check for user react on msg
+        async def check_reacts():
+            try:
+                check = (
+                    lambda r, u: r.message.id == conf_message.id
+                    and u == ctx.message.author
+                    and str(r.emoji) in emojis
+                )
+                react_emoji, _ = await ctx.bot.wait_for(
+                    "reaction_add", check=check, timeout=30
+                )
+                return react_emoji
+            except TimeoutError:
+                return None
+            finally:
+                await conf_message.delete()
+
+        # Add and check in parallel, so don't miss quick react
+        _, react_emoji = await asyncio.gather(add_reacts(), check_reacts())
+
+        # Get choice
         ind = emojis.index(str(react_emoji))
-        return rooms.get(ind)
+        return rooms[ind]
 
     def get_room_infos(self, room):
-        # Swap with campus map autocomplete for more reliability? but that need auth
-        map_req = req_or_none(f"https://search.warwick.ac.uk/api/maps?q={room}")
-        if map_req is None or not map_req.get("total"):
+        stripped = room.replace(".", "").replace(" ", "").lower()
+        if new := self.custom_room_names.get(stripped):
+            room = new
+        # Check Map Autocomplete API
+        map_req = req_or_none(
+            f"https://campus-cms.warwick.ac.uk//api/v1/projects/1/autocomplete.json?term={room}",
+            headers={"Authorization": "Token 3a08c5091e5e477faa6ea90e4ae3e6c3"},
```

Comment:
Is this token meant to be hardcoded and/or public?

Reply:
I think it may still be best to move to a variable in case it changes ever to make updating easier, but otherwise LGTM.